plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -633,57 +624,89 @@ def finalize(self):
+        and a repo, return a set of NEVRA tuples that can be used to filter out packages not
+        linked to that repo when generating a repo's updateinfo XML file.
+
+        :param repo_id: Repository ID for which to return the set of nevra
+        :type repo_id: str
+
+        :return: a set of NEVRA dicts for units in a single repo referenced by the given errata
+        :rtype: set of pulp_rpm.plugins.models.NEVRA tuples
+        """
+        repo_unit_nevra = set()
+        querysets = repo_controller.get_unit_model_querysets(repo_id, models.RPM)
+        nevra_scalars = itertools.chain(*[q.scalar(*models.NEVRA._fields) for q in querysets])
+        for scalar in nevra_scalars:
+            repo_unit_nevra.add(models.NEVRA(*scalar))
+        return repo_unit_nevra
+
+    def _get_pkglist_to_publish(self, erratum_unit):
         """
         Make a list of packages which will be listed in the published erratum.
 
         It is possible to filter packages only if `repo_unit_nevra` is available,
         otherwise publish everything except duplicated collections.
 
+        This merges multiple Pulp pkglists into a single pkglist for a given publication.
+
         :param erratum_unit: the erratum unit to analyze
         :param type: pulp_rpm.plugins.db.models.Errata
-        :return: list of pkglists to publish
-        :rtype: list of dicts
+        :return: pkglist to publish
+        :rtype: dict
         """
-        pkglists_to_publish = []
+        repo_id = self.get_repo().id
+        repo_unit_nevra = self._get_repo_unit_nevra(repo_id)
+        seen_packages = set()
+
+        # Build a new errata pkglist. Comments inline attempt to explain why it makes sense to
+        # explicitly use repo_id three times.
+        new_pkglist = {
+            # Due to a quirk in how yum merges pkglists in errata that appear in multiple
+            # repos, the pkglist collection name needs to be unique per-repo when published.
+            # repo_id is already unique and presumably validated, so use that. This is normally a
+            # human-readable string, like "A Human-Readable Repository Description". Pulp doesn't
+            # require storing that sort of string with a repo, so repo_id is the best we've got.
+            'name': repo_id,
+
+            # The repo "short name", like 'repo-name-short'. It also makes sense to use repo_id
+            # here. This appears to be unused by yum when parsing errata, but consistency with the
+            # "long" name field is good to have.
+            'short': repo_id,
+
+            # Pulp injects this value to indicate the repository a given pkglist was imported
+            # *from*. Since the next step is to filter all packages available in this errata known
+            # to pulp by all packages known to pulp, this should be set to the repo_id being
+            # published in case it's expected to exist in later steps.
+            '_pulp_repo_id': repo_id,
```

Comment:
I do not think `_pulp_repo_id` is used during publish step, only in errata merge logic during sync.